control/metrics.go

```diff
@@ -55,17 +55,28 @@ var (
 
 // hostnameReader, hostnamer created for mocking
 func init() {
-	hostnameReader = &hostnameReaderType{}
+	h, err := os.Hostname()
+	if err != nil {
+		log.WithFields(log.Fields{
+			"_module": "control",
+			"_file":   "metrics.go,",
+			"_block":  "addStandardAndWorkflowTags",
+			"error":   err.Error(),
+		}).Error("Unable to determine hostname")
+	}
+	hostnameReader = &hostnameReaderType{hostname: h}
```

Comment:
I think this is going to result in only calling once per run of snapd since init will only be run at that time, instead of the intended once per collect call.